platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+#include "coding/file_reader.hpp"
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+
+/// @return directory where binary resides, including slash at the end
+static bool GetBinaryFolder(string & outPath)
+{
+  int mib[4];
+  mib[0] = CTL_KERN;
+  mib[1] = KERN_PROC;
+  mib[2] = KERN_PROC_PATHNAME;
+  mib[3] = -1;
+  char path[4096];
+  size_t cb = ARRAY_SIZE(path);
+  if (::sysctl(mib, 4, path, &cb, NULL, 0) == 0) {
+    outPath = path;
+    outPath.erase(outPath.find_last_of('/') + 1);
+    return true;
+  }
+  return false;
+}
+
+Platform::Platform()
+{
+  // init directories
```

Comment:
All comments except inline should start from a capital letter and end with a dot.